docs/clients/gpio_controller.rst

```diff
@@ -0,0 +1,139 @@
+as well as the type of pull used. 
+Each pin set as an output may choose whether to output in a 
+`Push-Pull or Open-Drain configuration <https://ebics.net/the-difference-between-push-pull-output-and-open-drain-output-of-microcontroller-i-o-port/>`_.
+
+
+Arduino
+~~~~~~~
+
+To use the GPIO Controller in Arduino, ensure gpio_controller_client.hpp is included. This
+allows the creation of a GpioControllerClient object. See the Message Table below for available messages. 
+All message objects use the Short Name with a trailing underscore. 
+All messages use the standard Get/Set/Save functions. 
+
+A minimal working example for the GpioControllerClient is:
+
+.. code-block:: Arduino
+
+    #include <iq_module_communication.hpp>
+    
+    IqSerial ser(Serial2);
+    GpioControllerClient gpioController(0);
+    
+    void setup() {
+        ser.begin();
+        // Initialize Serial (for displaying information on the terminal)
+        Serial.begin(115200);
+    }
+    
+    void loop() {
+        int modeRegister = 0;
+        if(ser.get(gpioController.mode_register_, modeRegister))
+        Serial.println(modeRegister);
+    }
+
+C++
+~~~
+
+To use the GPIO Controller client in C++, include gpio_controller_client.hpp. This allows the
+creation of an GpioController object. See the Message Table below for available messages. All message objects
+use the Short Name with a trailing underscore. All messages use the standard Get/Set/Save functions.
+
+A minimal working example for the GpioControllerClient is:
+
+.. include:: ../notes/c-full-code.rst
+
+.. code-block:: c++
+
+    #include "generic_interface.hpp"
+    #include "gpio_controller_client.hpp"
+
+
+    void main(){
+        // Make a communication interface object
+        GenericInterface com;
+
+        // Make a GPIO Controller object with obj_id 0
+        GpioControllerClient gpioController(0);
+
+        // Use the GPIO Controller Client
+        gpioController.mode_register_.get(com)
+
+        // Insert code for interfacing with hardware here  
+    }
+
+Matlab
+~~~~~~
+
+To use the GPIO Controller client in Matlab, all Vertiq communication code must be included in your
+path. This allows the creation of a GpioControllerClient object. See the Message Table below for available messages.
+All message strings use the Short Names. All messages use the standard Get/Set/Save functions.
+
+A minimal working example for the GpioControllerClient is:
+
+.. code-block:: Matlab
+
+    % Make a communication interface object
+    com = MessageInterface(’COM18’,115200);
+    % Make a GpioControllerClient object with obj_id 0
+    gpioController = GpioControllerClient(’com’,com);
+    % Use the GpioControllerClient object
+    modeRegister = gpioController.get(’mode_register’);
+
+Python
+~~~~~~
+
+To use the GPIO Controller Client in Python, include ``iqmotion`` and create a module that has the GPIO Controller Client within it's firmware. 
```

Comment:
its not it's. same question about this already being within the fortiq docs. the fortiq has this client